### PR TITLE
HPCC-21981 Improve complete-uninstall to remove package hpccsystems-platform-with-spark.

### PR DESCRIPTION
**Pull request by gfortil**
…latform-with-spark.

Signed-off-by: Godson Fortil <Fortil.Godson@Lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [x] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [ ] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [x] Send notifications about my Pull Request position in Smoketest queue.
- [x] Test my draft Pull Request.

### CODE REVIEW

**Comment by gfortil** [comment]
@mckellyln please review.

File: initfiles/sbin/complete-uninstall.sh.in [hunk 1]
```diff
@@ -100,15 +100,17 @@ if [ -n "${lpath}" -a -h "${lpath}" ] ; then
 fi
 
 if [ -e /etc/debian_version ]; then
-    echo "Removing DEB"
-    apt-get remove -y hpccsystems-platform
+    pckname=$(apt-cache pkgnames hpccsystems-platform)
+    echo "Removing ${pckname}"
+    apt-get remove -y ${pckname}
```
Comment (mckellyln) [inline initfiles/sbin/complete-uninstall.sh.in]: Lets check that $pckname is not empty.

File: initfiles/sbin/complete-uninstall.sh.in [hunk 2]
```diff
@@ -100,15 +100,17 @@ if [ -n "${lpath}" -a -h "${lpath}" ] ; then
 fi
 
 if [ -e /etc/debian_version ]; then
-    echo "Removing DEB"
-    apt-get remove -y hpccsystems-platform
+    pckname=$(apt-cache pkgnames hpccsystems-platform)
+    echo "Removing ${pckname}"
+    apt-get remove -y ${pckname} 
     if [ $? -ne 0 ]; then
        echo "An error has occured during removal.  Try \`apt-get -f remove hpccsystems-platform\` to fix this issue."
```
Comment (mckellyln) [inline initfiles/sbin/complete-uninstall.sh.in]: Lets change hpccsystems-platform to be $pckname for this msg,

File: initfiles/sbin/complete-uninstall.sh.in [hunk 3]
```diff
@@ -100,15 +100,17 @@ if [ -n "${lpath}" -a -h "${lpath}" ] ; then
...
        exit 1
     fi
 elif [ -e /etc/redhat-release -o -e /etc/SuSE-release ]; then
-    echo "Removing RPM"
-    yum remove -y hpccsystems-platform
+    pckname=$(rpm -qa | grep hpccsystems-platform)
```
Comment (mckellyln) [inline initfiles/sbin/complete-uninstall.sh.in]: same here

File: initfiles/sbin/complete-uninstall.sh.in (hunk 2 above)
Comment (mckellyln) [reply initfiles/sbin/complete-uninstall.sh.in]: Or like below leave off the name as its echoed above.

**Comment by gfortil** [comment]
Making more changes

**Pull request closed without merging**